### PR TITLE
Switch from sphinx to use mkdocs 

### PR DESCRIPTION
**Pull request by ens-LCampbell**
- Initial set up configs
- Workflow update
- MD file creation
- auto docs gen_reference MD script
- Revert sphinx to backup and update build script
- Update project TOML for mkdocs/sphinx distinction

### CODE REVIEW

File: docs/gen_ref_pages.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+nav = mkdocs_gen_files.Nav()
+
+for path in sorted(Path("src").rglob("*.py")):
+    # print(f"Operating on path --> {path}")
+    module_path = path.with_suffix("")
+    # print(f"Module path --> {module_path}")
+    doc_path = path.relative_to("src").with_suffix(".md")
+    # print(f"Doc path --> {doc_path}")
+    full_doc_path = Path("reference", doc_path)
+    # print(f"Full Doc path --> {full_doc_path}")
```
Comment (JAlvarezJarreta) [inline docs/gen_ref_pages.py:16]:
```suggestion
    module_path = path.with_suffix("")
    doc_path = path.relative_to("src").with_suffix(".md")
    full_doc_path = Path("reference", doc_path)
```

File: docs/gen_ref_pages.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        doc_path = doc_path.with_name("index.md")
+        full_doc_path = full_doc_path.with_name("index.md")
+    elif parts[-1] == "__main__":
+        continue
+
+    nav[parts] = doc_path.as_posix()
+
+    with mkdocs_gen_files.open(full_doc_path, "w") as fd:  # 
+        identifier = ".".join(parts)  # 
+        print("::: " + identifier, file=fd)  #
```
Comment (JAlvarezJarreta) [inline docs/gen_ref_pages.py:31]:
```suggestion
    with mkdocs_gen_files.open(full_doc_path, "w") as fd:
        identifier = ".".join(parts)
        print("::: " + identifier, file=fd)
```

File: docs/gen_ref_pages.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    elif parts[-1] == "__main__":
+        continue
+
+    nav[parts] = doc_path.as_posix()
+
+    with mkdocs_gen_files.open(full_doc_path, "w") as fd:  # 
+        identifier = ".".join(parts)  # 
+        print("::: " + identifier, file=fd)  # 
+
+    mkdocs_gen_files.set_edit_path(full_doc_path, Path("../") / path)  #
```
Comment (JAlvarezJarreta) [inline docs/gen_ref_pages.py:33]:
```suggestion
    mkdocs_gen_files.set_edit_path(full_doc_path, Path("../") / path)
```

File: docs/gen_ref_pages.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    nav[parts] = doc_path.as_posix()
+
+    with mkdocs_gen_files.open(full_doc_path, "w") as fd:  # 
+        identifier = ".".join(parts)  # 
+        print("::: " + identifier, file=fd)  # 
+
+    mkdocs_gen_files.set_edit_path(full_doc_path, Path("../") / path)  # 
+
+with mkdocs_gen_files.open("reference/SUMMARY.md", "w") as nav_file:  # 
+    nav_file.writelines(nav.build_literate_nav())  #
```
Comment (JAlvarezJarreta) [inline docs/gen_ref_pages.py:36]:
```suggestion
with mkdocs_gen_files.open("reference/SUMMARY.md", "w") as nav_file:
    nav_file.writelines(nav.build_literate_nav())
```

**Review by JAlvarezJarreta — APPROVED** [review]
I would personally drop `docs/sphinx/source`, `scripts/setup/docs/build_sphinx_docs.sh` and the `sphinx` label in pyproject, but besides that great addition :scroll: I have done just a few minor suggestions on code cleanup and such.

I have a feeling some of these documents could be pulled from their original source to be generated/updated, e.g. `license.md`, but that can be done in a later stage (if needed).

_Note:_ unless it is required to be this way, `mkdocs.yaml` should be `mkdocs.yml`

File: mkdocs.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# mkdocs.yml
+site_name: Ensembl-GenomIO (v0.1)
```
[the remark this answers is not in the document]
Comment (ens-LCampbell) [reply mkdocs.yaml]: I've removed the versioning information, changed the name to `Ensembl GenomIO`. Also removed Sphinx legacy scripts, source etc as suggested. If we can get from git history then fine by me.

File: docs/index.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+	        │       └── validate.py
+	        ├── seq_region
+	        │   ├── dump.py
+	        │   └── prepare.py
+	        └── utils
+	            ├── archive_utils.py
+	            └── json_utils.py
+
+## License 
+Software as part of [Ensembl GenomIO](https://github.com/Ensembl/ensembl-genomio) is distributed under the [Apache-2.0 License](https://www.apache.org/licenses/LICENSE-2.0.txt)
```
Comment (JAlvarezJarreta) [inline docs/index.md:94]:
```suggestion
Software as part of [Ensembl GenomIO](https://github.com/Ensembl/ensembl-genomio) is distributed under the [Apache-2.0 License](https://www.apache.org/licenses/LICENSE-2.0.txt).
```

File: docs/usage.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+```
+python3.7 -m venv path/to/virtual_env
+. path/to/virtual_env/bin/activate
+pip install -e .
+```
+
+*Do not forget to load this environment by sourcing `virtual_env/bin/activate` in order to run ensembl-genomio pipeline(s).*
+
+
+
```
Comment (JAlvarezJarreta) [inline docs/usage.md:18]:
```suggestion
```

**Review by JAlvarezJarreta — APPROVED** [review]
Looks quite neat :eyes:

**Pull request merged**